onnxruntime/core/providers/webnn/builders/model.cc

```diff
@@ -60,24 +58,23 @@ Status Model::Predict(const InlinedHashMap<std::string, OnnxTensorData>& inputs,
     }
     auto num_elements = SafeInt<size_t>(Product(tensor.tensor_info.shape));
     emscripten::val view{emscripten::typed_memory_view(num_elements, static_cast<const float*>(tensor.buffer))};
+#ifdef ENABLE_WEBASSEMBLY_THREADS
     val_vec.insert({name, view});
-    emscripten::val non_shared_array = emscripten::val::global("Float32Array").new_(static_cast<uint32_t>(num_elements));
-    if (SharedArrayBuffer.as<bool>()) {
-      wnn_outputs_.set(name, non_shared_array);
-    } else {
-      wnn_outputs_.set(name, view);
-    }
+    wnn_outputs_.set(name, view.call<emscripten::val>("slice"));
```

Comment:
As I commented before, you may want to pre-allocate the output buffers for `wnn_outputs_`. Calling `view.slice()` for each compute is not necessary, because it copies the content of `view` to `wnn_outputs_` which is not needed.